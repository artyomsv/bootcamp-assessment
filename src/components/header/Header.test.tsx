import * as React from 'react';
import {createShallow} from '@material-ui/core/test-utils';
import {createMount} from '@material-ui/core/test-utils';
import Header from './Header.widget';
import IconButton from '@material-ui/core/IconButton';

const shallow = createShallow();
const mount = createMount();

describe('renders without crashing', () => {
  test('if renders', () => {
    const mockFn = jest.fn();
    const result = shallow(<Header onMenuClick={mockFn}/>);
    expect(result).toMatchSnapshot();
  });
});


describe('Test Header', () => {
  test('if renders', () => {
    const mockFn = jest.fn();
    const wrap = mount(<Header onMenuClick={mockFn}/>);
    expect(wrap.find(IconButton).length).toBe(1);

    wrap.find(IconButton).simulate('click');
    expect(mockFn).toHaveBeenCalledTimes(1);
  });
});