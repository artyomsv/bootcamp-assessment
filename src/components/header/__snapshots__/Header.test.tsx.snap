// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing if renders 1`] = `
<Header
  classes={
    Object {
      "appBar": "Header-appBar-1",
      "flex": "Header-flex-3",
      "menuButton": "Header-menuButton-4",
      "root": "Header-root-2",
    }
  }
  onMenuClick={[MockFunction]}
/>
`;
